unified-runtime/source/adapters/opencl/adapter.hpp

```diff
@@ -7,18 +7,22 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
+#pragma once
+
 #include "logger/ur_logger.hpp"
 #include "platform.hpp"
 
 #include "CL/cl.h"
+#include "common.hpp"
 #include "logger/ur_logger.hpp"
 
 struct ur_adapter_handle_t_ {
   ur_adapter_handle_t_();
+  ~ur_adapter_handle_t_();
```

Comment:
I've been adding a deleted copy constructor alongside non-trivial destructor declarations out of a vague sense of best practise